src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -124,8 +124,35 @@ def set_package_assessment_status(self, package_names, package_versions, classif
                 }
                 self.__assessment_packages.append(record)
 
+        self.__assessment_packages = self.sort_packages_by_classification_and_state(self.__assessment_packages)
         self.set_assessment_substatus_json()
 
+    def sort_packages_by_classification_and_state(self, packages_list):
+        """ Sorts a list of packages (usually either self.__assessment_packages or self.__installation_packages) by classification and patchState properties.
+            (sorting order from highest priority to lowest):
+            1. Classification: Security, Critical, Other, Unclassified
+            2. Patch Installation State: Failed, Installed, Available, Pending, Excluded, NotSelected
+        """
+        def sort_patch_state_key(x):
+            # Only for installation result packages
+            if "patchInstallationState" in x.keys():
+                return Constants.PatchStateOrder[x["patchInstallationState"]]
+            else:
+                return 0
+
+        def sort_classification_key(x):
+            lowest_classification = Constants.PackageClassificationOrder[x["classifications"][0]]
+            for i in range(1, len(x["classifications"])):
+                lowest_classification = min(lowest_classification, Constants.PackageClassificationOrder[x["classifications"][i]])
+            return lowest_classification
+
+        # Sort by patch state first then sort by classification so each type of classification is already sorted at the end
+        list_sorted_by_patch_state = sorted(packages_list, key=sort_patch_state_key)
+        list_sorted_by_classification = sorted(list_sorted_by_patch_state, key=sort_classification_key)
+        # COMMENT: these sort calls can be combined into one by separating the value ranges in the tables above, such as classification_order to 100, 200, 300, etc
+        # and having one key function that combines the result of classification_order and patch_state_order
```

Comment:
Check note: Is the outcome protected by a UT given dependency on specific behavior of sorted

Reply:
Yes

Reply:
Verified in principle - closing comment.